applicationset/controllers/applicationset_controller_test.go

```diff
@@ -1320,8 +1353,10 @@ func TestRemoveFinalizerOnInvalidDestination_DestinationTypes(t *testing.T) {
 				},
 			}
 
-			objects := append([]runtime.Object{}, secret)
-			kubeclientset := kubefake.NewSimpleClientset(objects...)
+			kubeclientset := getDefaultTestClientSet()
+			err := kubeclientset.Tracker().Add(secret)
```

Comment:
Instead of accessing the tracker like this, maybe just parameterize `getDefaultTestClientSet` to accept additional objects? I've seen that pattern in other tests.